src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+
+const useFetchPostNotificationHook = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+
+  const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
+    if (!localDb) return;
+    const allPromises: Promise<void>[] = notifications.map((notification) => {
+      const channelList = fromNotificationAPI(notification);
+      return channelList.saveIfLatest(localDb);
+    });
+
+    try {
+      await Promise.all(allPromises);
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving notifications:', e);
+    }
+  };
+
+  const getAllSignedPostNotifications = async () => {
+    try {
+      const signedPostNotifications = await SignedMessageRepo.getAllSignedPostNotifications();
+      saveNotifications(signedPostNotifications, ChannelList.fromSignedPostNotificationAPI);
+    } catch (e) {
+      console.log('error on getting signedPostNotifications:', e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getAllSignedPostNotifications` is declared as async but it doesn't await the `saveNotifications` function call at line 28. This could lead to unexpected behavior because `saveNotifications` is an asynchronous operation and may not complete before the next lines of code are executed.

```diff
      const signedPostNotifications = await SignedMessageRepo.getAllSignedPostNotifications();
-     saveNotifications(signedPostNotifications, ChannelList.fromSignedPostNotificationAPI);
+     await saveNotifications(signedPostNotifications, ChannelList.fromSignedPostNotificationAPI);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->